<a href="https://colab.research.google.com/github/shivanisisodia7/Dash-by-Plotly/blob/master/Data_Analysis_Final_Assignment_Shivani_Singh.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Task 1: Importing Libraries , and Uploading Data

In [280]:
# Importing required libraries
import pandas as pd
import re
import numpy as np

In [281]:
# data from the 2018,2019 and 2020
data_18 = pd.read_csv("https://raw.githubusercontent.com/dphi-official/Datasets/master/IT_Salary_Survey_EU_18-20/Survey_2018.csv")
data_19 = pd.read_csv("https://raw.githubusercontent.com/dphi-official/Datasets/master/IT_Salary_Survey_EU_18-20/Survey_2019.csv")
data_20 = pd.read_csv("https://raw.githubusercontent.com/dphi-official/Datasets/master/IT_Salary_Survey_EU_18-20/Survey_2020.csv")

## Task 2: Data Analysis


In [282]:
# Display the first 5 rows of 2018 data
data_18.head()

Timestamp   Age  ... Company size   Company type
0  14/12/2018 12:41:33  43.0  ...     100-1000        Product
1  14/12/2018 12:42:09  33.0  ...       50-100        Product
2  14/12/2018 12:47:36  32.0  ...        1000+        Product
3  14/12/2018 12:50:15  25.0  ...        1000+        Product
4  14/12/2018 12:50:31  39.0  ...     100-1000  Ecom retailer

[5 rows x 14 columns]

In [283]:
# Summary of the data
data_20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1253 entries, 0 to 1252
Data columns (total 23 columns):
 #   Column                                                                                                                   Non-Null Count  Dtype  
---  ------                                                                                                                   --------------  -----  
 0   Timestamp                                                                                                                1253 non-null   object 
 1   Age                                                                                                                      1226 non-null   float64
 2   Gender                                                                                                                   1243 non-null   object 
 3   City                                                                                                                     1253 non-null   o

## observations from the above data

* In this data we have 1253 rows and 23 columns

* We have 19 Objects ( String ) columns  and 4 Float ( Numerical with Decimal)  Data type columns

* All those columns which have 1253 non-null count do not have any null values and rest of the columns have some null values in it, which will need to be filled either with Mean, mode or need to be deleted, this is totally dependent on the other factors as well like type of data and how crucial the information that particular row contains.


In [284]:
# Descriptive Statistics of 2018 data
data_18.describe()

Age  ...  Salary two years ago
count  672.000000  ...            463.000000
mean    32.183036  ...          58013.475162
std      5.107268  ...          20413.048908
min     21.000000  ...          10001.000000
25%     29.000000  ...          48000.000000
50%     32.000000  ...          56000.000000
75%     35.000000  ...          67000.000000
max     60.000000  ...         150000.000000

[8 rows x 5 columns]

In [285]:
# Missing Values in 2018 data
data_18.isnull().sum()

Timestamp                               0
Age                                    93
Gender                                 14
City                                   29
Position                               28
Years of experience                    33
Your level                             22
Current Salary                         15
Salary one year ago                   169
Salary two years ago                  302
Are you getting any Stock Options?     23
Main language at work                  15
Company size                           15
Company type                           35
dtype: int64

In [286]:
# How many people responded to the survey in each of the 3 years? Has the number increased or decreased over the years?
print("No.of People responded during 2018 ",len(data_18))
print("No.of People responded during 2019 ",len(data_19))
print("No.of People responded during 2020 ",len(data_20))

difference_18_to_19 = "Percentage increase from 2018 to 2019: {:.0%}".format((len(data_19)-len(data_18))/len(data_18))
difference_19_to_20 = "Percentage increase from 2019 to 2020: {:.0%}".format((len(data_20)-len(data_19))/len(data_19))



print(difference_18_to_19)
print(difference_19_to_20)

No.of People responded during 2018  765
No.of People responded during 2019  991
No.of People responded during 2020  1253
Percentage increase from 2018 to 2019: 30%
Percentage increase from 2019 to 2020: 26%


There is increase in number of people responded in the Survey but if we compare with respect to previous we are seeing only 26% increase however during the year 2018 to 2019 we have seen 30% increase in overall numbers

In [287]:
# Unique values in Number of Vacation days column of 2020 data and their frequency

data_20.groupby("Number of vacation days")["Number of vacation days"].count().sort_values(ascending = False)


Number of vacation days
30                                              488
28                                              233
27                                              102
25                                               91
26                                               71
24                                               67
29                                               24
20                                               13
21                                               10
31                                                8
32                                                8
22                                                8
36                                                5
35                                                5
40                                                4
23                                                4
0                                                 4
45                                                3
33                                      

In [288]:
data_20[data_20['Number of vacation days'] == 'unlimited']

Timestamp  ...  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR
324  24/11/2020 17:24:17  ...                                               1500                                                                      
771  26/11/2020 14:48:23  ...                                                500                                                                      
961  30/11/2020 13:50:56  ...                                                NaN                                                                      

[3 rows x 23 columns]

## Observations

* 30 Days vacation has been provided to the maximum employees which is 488

* After considering both of the above information it came to notice that unlimited vacation days have been provided to only those employees who are part of unlimited employment contract

## Task 3: Data Cleaning

## Data 2020 Cleaning
* Rename column name
* Check the null values
* Imputing values in null columns

In [289]:
# Rename the column Position to "Position" with out Space
data_20.rename(columns={'Position ':'Position'},inplace=True)

In [290]:
# Check for missing value in 2020 data for all columns
data_20.isnull().sum()

Timestamp                                                                                                                    0
Age                                                                                                                         27
Gender                                                                                                                      10
City                                                                                                                         0
Position                                                                                                                     6
Total years of experience                                                                                                   16
Years of experience in Germany                                                                                              32
Seniority level                                                                                                

In [291]:
# Fill the null values - Mean for Numerical values and Mode for Categorical values
cat_cols_20 = data_20.select_dtypes(include='object')
num_cols_20 = data_20.select_dtypes(exclude='object')

data_20_cat = cat_cols_20.apply(lambda x:x.fillna(x.value_counts().index[0]))
data_20_num = num_cols_20.fillna(num_cols_20.mean())


In [292]:
# Merge the both of the data 
data_20_1 = pd.concat([data_20_cat,data_20_num],axis=1)

In [293]:
# Check if their shape is same or not
print("Previous Data Shape",data_20.shape)
print("New Data Shape",data_20_1.shape)


# Check if there is still any null value left in the data
print("Checking if still there is any null value left in the data: \n ")
data_20_1.isnull().sum()

Previous Data Shape (1253, 23)
New Data Shape (1253, 23)
Checking if still there is any null value left in the data: 
 


Timestamp                                                                                                                  0
Gender                                                                                                                     0
City                                                                                                                       0
Position                                                                                                                   0
Total years of experience                                                                                                  0
Years of experience in Germany                                                                                             0
Seniority level                                                                                                            0
Your main technology / programming language                                                                                0


## Data 19 Cleaning
* Rename column name
* Check the null values
* Imputing values in null columns

In [294]:
# Check if there is any null value in 2019 data
data_19.isnull().sum()

Zeitstempel                                                                                               0
Age                                                                                                     109
Gender                                                                                                    0
City                                                                                                      0
Seniority level                                                                                          15
Position (without seniority)                                                                              1
Years of experience                                                                                       0
Your main technology / programming language                                                              14
Yearly brutto salary (without bonus and stocks)                                                           1
Yearly bonus                

In [295]:
# Separate the Object and Numerical data 
cat_cols_19 = data_19.select_dtypes(include='object')
num_cols_19 = data_19.select_dtypes(exclude='object')

data_19_cat = cat_cols_19.apply(lambda x:x.fillna(x.value_counts().index[0]))
data_19_num = num_cols_19.fillna(num_cols_19.mean())


In [296]:
# Merge the both of the data 
data_19_1 = pd.concat([data_19_cat,data_19_num],axis=1)

In [297]:
# Check if their shape is same or not
print("Previous Data Shape",data_19.shape)
print("New Data Shape",data_19_1.shape)


# Check if there is still any null value left in the data
print("Checking if still there is any null value left in the data: \n ")
data_19_1.isnull().sum()

Previous Data Shape (991, 23)
New Data Shape (991, 23)
Checking if still there is any null value left in the data: 
 


Zeitstempel                                                                                               0
Gender                                                                                                    0
City                                                                                                      0
Seniority level                                                                                           0
Position (without seniority)                                                                              0
Your main technology / programming language                                                               0
Main language at work                                                                                     0
Company name                                                                                              0
Company size                                                                                              0
Company type                

* That 0 column will be deleted further as this column is of no use 

## 2018 Data Cleaning
* Rename column name
* Check the null values
* Imputing values in null columns

In [298]:
# Check if there is any null value in 2019 data
data_18.isnull().sum()

Timestamp                               0
Age                                    93
Gender                                 14
City                                   29
Position                               28
Years of experience                    33
Your level                             22
Current Salary                         15
Salary one year ago                   169
Salary two years ago                  302
Are you getting any Stock Options?     23
Main language at work                  15
Company size                           15
Company type                           35
dtype: int64

In [299]:
# Separate the Object and Numerical data 
cat_cols_18 = data_18.select_dtypes(include='object')
num_cols_18 = data_18.select_dtypes(exclude='object')

data_18_cat = cat_cols_18.apply(lambda x:x.fillna(x.value_counts().index[0]))
data_18_num = num_cols_18.fillna(num_cols_18.mean())

In [300]:
# Merge the both of the data 
data_18_1 = pd.concat([data_18_cat,data_18_num],axis=1)

In [301]:
# Check if their shape is same or not
print("Previous Data Shape",data_18.shape)
print("New Data Shape",data_18_1.shape)


# Check if there is still any null value left in the data
print("Checking if still there is any null value left in the data: \n ")
data_18_1.isnull().sum()

Previous Data Shape (765, 14)
New Data Shape (765, 14)
Checking if still there is any null value left in the data: 
 


Timestamp                             0
Gender                                0
City                                  0
Position                              0
Your level                            0
Are you getting any Stock Options?    0
Main language at work                 0
Company size                          0
Company type                          0
Age                                   0
Years of experience                   0
Current Salary                        0
Salary one year ago                   0
Salary two years ago                  0
dtype: int64

## Remove Irrelevant columns from the data

In [302]:
# remove the timestamp column from all the three data
data_18_1.drop("Timestamp",axis=1,inplace=True)
data_19_1.drop(["Zeitstempel","0"],axis=1,inplace=True)
data_20_1.drop("Timestamp",axis=1,inplace=True)

## Task 4: Data Visualization using Plotly



In [303]:
# Import plotly and Dash library
import plotly
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go

* **Create a pie chart to analyze the Company types in the year 2019. Are Consulting / Agency companies more popular than Startups ?**

In [304]:
# Grouping by the data with respect to company type and we will consider most count type as the most popular
company_type = data_19_1.groupby('Company type')['Company type'].count()
company_type

Company type
Bank                      6
Bodyshop / Outsource     30
Consulting / Agency     117
Outsource                 1
Product                 650
Startup                 181
University                6
Name: Company type, dtype: int64

In [332]:
# Create a pie chart to analyze the Company types in the year 2019. Are Consulting / Agency companies more popular than Startups?
fig = px.pie(data_frame=company_type,values='Company type',names= company_type.index,title= "Type of Company and their Ratio")
fig.update_traces(textinfo="label + percent",insidetextfont=dict(color = "white"))
fig.show()

We have summarized the data on the basis of Company type and concluded the popularity on the basis of number of people filled this Survey 

We can see from the above pie chart that Consulting/Agency has only 11% people who filled this form on the contrary from startup 18% people have filled this survey so it means startup has more popularity than Agency/Consultancy.

* **Create a line plot of the Total years of experience vs the current salary(taking the median salary for each of the different experience years) of the year 2018**

In [306]:

salary = data_18_1.groupby("Years of experience")["Current Salary"].median()
salary

Years of experience
0.000000      50000.000000
0.500000      46000.000000
1.000000      46000.000000
1.500000      58000.000000
2.000000      50000.000000
2.500000      50000.000000
3.000000      55555.000000
4.000000      60000.000000
4.500000      39150.000000
5.000000      60000.000000
6.000000      64000.000000
7.000000      68000.000000
7.500000      65000.000000
8.000000      70000.000000
8.548497      68381.765333
9.000000      70000.000000
10.000000     70000.000000
11.000000     71000.000000
12.000000     71000.000000
13.000000     73500.000000
14.000000     71500.000000
15.000000     68000.000000
16.000000     90500.000000
17.000000     80000.000000
18.000000     72500.000000
19.000000     84000.000000
20.000000     80000.000000
21.000000     90000.000000
25.000000    109000.000000
30.000000     64000.000000
38.000000     58000.000000
Name: Current Salary, dtype: float64

In [333]:
fig = px.line(salary,x=salary.index,y="Current Salary",title="Average Salary with respect to Experience")
fig.show()

* Above graph shows the more the experience one have, more will be the salary the other person will get, but there is one thing which is not highlighted here is the designation / Type of work because these two factors affect the salary as well 
* we are seeing the highest salary is given to person who has 25 years of experience and then graph starts to fall which might have occured due to designation / knowledge one person possess and other factors.

* **Create the above plot again and add 2 more line plots to the same graph, that display the Total years of experience vs the median Yearly brutto salary (without bonus and stocks) of the year 2019 and 2020**

In [308]:
# 2019 Data
brutto_19 = data_19_1.groupby("Years of experience")["Yearly brutto salary (without bonus and stocks)"].median()
brutto_19

Years of experience
0      55000.0
1      46800.0
2      52500.0
3      55000.0
4      62000.0
5      65000.0
6      68000.0
7      67000.0
8      74500.0
9      74500.0
10     76500.0
11     76500.0
12     77700.0
13     70000.0
14     80000.0
15     75000.0
16     72000.0
17     75000.0
18     82500.0
19     73000.0
20     78000.0
22    100500.0
24    115000.0
25     79800.0
28    110000.0
30     60000.0
Name: Yearly brutto salary (without bonus and stocks), dtype: float64

In [309]:
# Cleaning for 2020 Data

experience = []

# Normalize Data for exp
for i in data_20_1['Total years of experience']:
  if i == '1,5':
    experience.append('1.5')
  elif i == '1 (as QA Engineer) / 11 in total':
    experience.append('11')
  elif i == '2,5':
    experience.append('2.5')
  elif i == '15, thereof 8 as CTO':
    experience.append('15')
  elif i == '6 (not as a data scientist, but as a lab scientist)':
    experience.append('6')
  elif i == '383':
    experience.append('38')
  elif i == 'less than year':
    experience.append('0.5')
  else:
    experience.append(i)


# Change the List to series 
exp = pd.Series([x for x in experience])


# Merge the data with the new column
data_20_1['Experience'] = exp.astype(float)



brutto_20 = data_20_1.groupby("Experience")["Yearly brutto salary (without bonus and stocks) in EUR"].median()

In [310]:
fig = px.line(salary,x=salary.index,y="Current Salary")
fig = px.line(brutto_19,x=brutto_19.index,y="Yearly brutto salary (without bonus and stocks)")
fig = px.line(brutto_20,x=brutto_20.index,y="Yearly brutto salary (without bonus and stocks) in EUR")
fig.update_layout(title="Average Brutto Salary",xaxis_title = 'Years of Experience',yaxis_title='Brutto Salary')


fig.show()

In [311]:
# Now, create the above plot again and add 2 more line plots to the same graph, 
# that display the Total years of experience vs the median Yearly brutto salary (without bonus and stocks) of the year 2019 and 2020

fig = go.Figure()

fig.add_trace(go.Scatter(x=salary.index,y=salary.values,name='2018 Current Salary',mode = 'markers',marker_size=10))
fig.add_trace(go.Scatter(x=brutto_19.index,y=brutto_19.values,name='2019 Brutto Salary',mode='markers',marker_size=10))
fig.add_trace(go.Scatter(x=brutto_20.index,y=brutto_20.values,name='2020 Brutto Salary',mode='markers',marker_size=10))
fig.update_layout(title="Average Brutto Salary",xaxis_title = 'Years of Experience',yaxis_title='Brutto Salary')


fig.show()

* **Create a bar chart to analyse the popularity of the main technology/ programming languages amongst the respondents in the year 2020. Which technology is the most popular? Which technology is the least popular (with less than 4 responses)**

In [ ]:
# Check the uniques in Main Technology and Programming Language column
data_20_1['Your main technology / programming language'].unique()

In [313]:
# Copied data only for technology portion
technology_data = data_20_1.copy()


In [314]:
# Change the names of the same format
technology_data['Your main technology / programming language'] = technology_data['Your main technology / programming language'].str.upper()

In [315]:
language = []

for i in technology_data['Your main technology / programming language']:
  if i == "JAVASCRIPT":
    language.append("JAVA")
  elif i == "PYRHON":
    language.append("PYTHON")
  elif i == "PYTORCH":
    language.append("PYTHON")
  elif i == "PYTHON (DJANGO)":
    language.append("PYTHON")
  elif i == "PYTHIN":
    language.append("PYTHON")
  elif i == "JAVSCRIPT":
    language.append("JAVA")
  elif i == "PLOTLY DASH":
    language.append("PYTHON")
  elif i == "FLASK":
    language.append("PYTHON")
  elif i == "DJANGO":
    language.append("PYTHON")
  elif i == "ML/PYTHON":
    language.append("PYTHON")
  elif i == "PYTHON/NLP":
    language.append("PYTHON")
  elif i == "PYTHON, PYTORCH":
    language.append("PYTHON")
  else:
    language.append(i.strip())

technology_data['language'] = pd.Series([x for x in language])


In [316]:
# Delete rows with None and other irrelavant data
tech_data = technology_data.drop(technology_data[(technology_data["language"] == 'NONE') & (technology_data["language"] == 'NO THING') & (technology_data["language"] == '--')].index)

**Famous Technology - only those for which we got response more than 10**

In [317]:
data = tech_data.groupby("language")["language"].count().sort_values(ascending = False)

top_10_data = data[data > 10]

top_10_data

language
JAVA          397
PYTHON        218
PHP            71
C++            41
C#             31
SWIFT          26
KOTLIN         25
GO             25
SCALA          25
TYPESCRIPT     24
.NET           23
RUBY           19
SQL            18
JS             15
Name: language, dtype: int64

In [318]:
# Bar chart for Top 10 Technologies
fig = px.bar(top_10_data,x = top_10_data.index,y="language",title="Top 10 Languages",color="language")
fig.update_layout(xaxis_title = 'Programming languages',yaxis_title = 'No of People Use this langauge')
fig.show()


* **Least Popular ( Less than 4 responses)**

In [319]:
last_4_data = data[data < 4]

In [320]:
fig = px.bar(last_4_data,x = last_4_data.index,y="language",title="Least Use Languages",color="language")
fig.update_layout(xaxis_title = 'Programming languages',yaxis_title = 'No of People Use this langauge')
fig.show()


* **Create a pie plot indicating the gender ratio of the respondents in the year 2020.**

In [321]:
gender_ratio = data_20_1.groupby("Gender")["Gender"].count()
gender_ratio

Gender
Diverse       2
Female      192
Male       1059
Name: Gender, dtype: int64

In [322]:
fig = px.pie(data_frame=gender_ratio,values='Gender',names= gender_ratio.index,title="Gender Ratio in 2020 Data")
fig.update_traces(textinfo="label + percent",insidetextfont=dict(color = "white"))
fig.show()

* it seems like a Males are more than Females and Diverse are the ones which chose not to specify their gender

## Additional Analysis 

* Lost Job due to Covid - In Proportion of Gender
* Lost Job due to Covid - With respect to Seniority Level


In [323]:
# How many people have lost their job due to coronavirus and which gender has lost the most

covid_outbreak = data_20_1[["Gender","Seniority level","Have you lost your job due to the coronavirus outbreak?"]]

covid_outbreak.head()

Gender  ... Have you lost your job due to the coronavirus outbreak?
0   Male  ...                                                 No     
1   Male  ...                                                 No     
2   Male  ...                                                Yes     
3   Male  ...                                                 No     
4   Male  ...                                                 No     

[5 rows x 3 columns]

In [324]:
lost_job = []

for i in covid_outbreak["Have you lost your job due to the coronavirus outbreak?"]:
  if i == "No":
    lost_job.append(0)
  else:
    lost_job.append(1)


covid_outbreak["Lost Job"] = pd.Series([x for x in lost_job]).astype(int)


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:10: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [325]:
fig = px.pie(data_frame=covid_outbreak,values='Lost Job',names= "Gender",title="Lost Job due to Covid with respect to Gender")
fig.update_traces(textinfo="label + percent",insidetextfont=dict(color = "white"))
fig.show()

* As we can see Males are higher in percentage as compare to the other gender

In [326]:
# Unique Names in the Seniority Level
covid_outbreak["Seniority level"].unique()

array(['Senior', 'Lead', 'Junior', 'Middle', 'Head', 'No level', 'VP',
       'Manager', 'Work Center Manager', 'CTO', 'No level ', 'Director',
       'Key', 'C-level executive manager', 'Principal', 'intern',
       'Student', 'no idea, there are no ranges in the firm ', 'C-Level',
       'Working Student', 'Entry level', 'Intern', 'student',
       'Self employed'], dtype=object)

Student,Junior , Middle and High

Lead = Middle
Head = High
No Level = Junior
VP = High
Manager = High
Work Center Manager = High
CTO = High
Director = High
Key = Middle
C-Level Executive manager = Middle
Principal = High
Intern  = Junior
Student = Delete it
No-idea = Junior
C-level = Junior
Working Sudent = Junior
Entry Level = Junior
Intern = Junior
Student = Delete
Self employed = Delete


In [327]:
covid_outbreak_2 = covid_outbreak[["Seniority level","Lost Job"]]

covid_outbreak_2

Seniority level  Lost Job
0             Senior         0
1             Senior         0
2               Lead         1
3             Junior         0
4             Senior         0
...              ...       ...
1248          Senior         1
1249          Senior         0
1250            Lead         0
1251          Middle         0
1252          Middle         0

[1253 rows x 2 columns]

In [328]:
# Changes in Seniority Level column - Normalize the column Name

level = []

for i in covid_outbreak_2["Seniority level"]:

  if i == "Senior":
    level.append("High")
  elif i == "Lead":
    level.append("Middle")
  elif i == "Head":
    level.append("High")
  elif i == "No level":
    level.append("Junior")
  elif i == "VP":
    level.append("High")
  elif i == "Manager":
    level.append("High")
  elif i == "Work Center Manager":
    level.append("High")
  elif i == "CTO":
    level.append("High")
  elif i == "No level ":
    level.append("Junior")
  elif i == "Director":
    level.append("High")
  elif i == "Key":
    level.append("Middle")
  elif i == "C-level executive manager":
    level.append("Middle")
  elif i == "Principal":
    level.append("High")
  elif i == "intern":
    level.append("Junior")
  elif i == "Student":
    level.append("Junior")
  elif i == "no idea, there are no ranges in the firm ":
    level.append("Junior")
  elif i == "C-Level":
    level.append("Middle")
  elif i == "Working Student":
    level.append("Junior")
  elif i == "Entry level":
    level.append("Junior")
  elif i == "Intern":
    level.append("Junior")
  elif i == "student":
    level.append("Junior")
  else:
    level.append(i)



covid_outbreak_2["level"] = pd.Series([x for x in level]) 


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:54: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [329]:
# Delete the self employed
self_employed = covid_outbreak_2[covid_outbreak_2["level"] == 'Self employed'].index

covid_outbreak_2.drop(self_employed,inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:4174: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [330]:
fig = px.pie(data_frame=covid_outbreak_2,values='Lost Job',names= "level",title="Lost Job due to Covid with respect to Seniority level")
fig.update_traces(textinfo="label + percent",insidetextfont=dict(color = "white"))
fig.show()

* Higher level people have lost their job due to covid followed by Middle then Junior